linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_commonauth/rule.yml

```
@@ -0,0 +1,64 @@
+documentation_complete: true
+
+prodtype: sle15
+
+title: "Set PAM's Common Authentication Hashing Algorithm"
+
+description: |-
+    The PAM system service can be configured to only store encrypted
+    representations of passwords. In
+    <tt>/etc/pam.d/common-auth</tt>,
+    the
+    <tt>auth</tt> section of the file controls which PAM modules execute
+    during a password change. Set the <tt>pam_unix.so</tt> module in the
+    <tt>auth</tt> section to include the argument <tt>sha512</tt>, as shown
+    below:
+    <br />
+    <pre>auth   required    pam_unix.so sha512 <i>other arguments...</i></pre>
+    <br />
+    This will help ensure when local users change their authentication method,
+    hashes for the new authentications will be generated using the SHA-512
+    algorithm. This is the default.
+
+rationale: |-
+    Unapproved mechanisms used for authentication to the cryptographic module
+    are not verified and therefore cannot be relied on to provide
+    confidentiality or integrity, and data may be compromised.
+    This setting ensures user and group account administration utilities are
+    configured to store only encrypted representations of passwords.
+    Additionally, the <tt>crypt_style</tt> configuration option ensures the use
+    of a strong hashing algorithm that makes password cracking attacks more
+    difficult.
+
+severity: medium
+
+identifiers:
+    cce@sle15: CCE-85754-0
+
+references:
+    disa@sle15: CCI-000803
+    nist@sle15: IA-7,IA-7.1
+    srg@sle15: SRG-OS-000120-GPOS-00061
+    vmmsrg@sle15: SRG-OS-000480-VMM-002000
+    stigid@sle15: SLES-15-010250
+
+ocil_clause: 'it does not'
+
+ocil: |-
+    Inspect the contents of <tt>/etc/pam.d/common-auth</tt>
+    and ensure that the <tt>pam_unix.so</tt> module includes the argument
+    <tt>sha512</tt>:
+    <pre>$ grep sha512 /etc/pam.d/common-auth</pre>
+
+platform: pam
+
+template:
+    name: pam_options
+    vars:
+      path: /etc/pam.d/common-auth
+      type: password
```

I suppose you want to check auth section, not password section.
```suggestion
      type: auth
```

---

Thanks :bow:, should be ok in edeb2c5